src/test/java/org/takes/servlet/HttpServletRequestFakeTest.java

```diff
@@ -0,0 +1,105 @@
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.takes.servlet;
+
+import java.util.Collections;
+import java.util.NoSuchElementException;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.hamcrest.core.IsEqual;
+import org.junit.Rule;
+import org.junit.jupiter.api.Test;
+import org.junit.rules.ExpectedException;
+import org.takes.rq.RqFake;
+import org.takes.rq.RqWithHeaders;
+
+/**
+ * Test case for {@link  HttpServletRequestFake}.
+ *
+ * @since 1.15
+ */
+public final class HttpServletRequestFakeTest {
+    /**
+     * A rule for handling an exception.
+     */
+    @Rule
+    public final ExpectedException exception = ExpectedException.none();
+
+    @Test
+    public void headersNotFound() {
+        this.exception.expect(NoSuchElementException.class);
+        this.exception.expectMessage("Value of header foo not found");
+        new HttpServletRequestFake(
+            new RqWithHeaders(
+                new RqFake(),
+                "MyHeader: theValue",
+                "MyOtherHeader: aValue"
+            )
+        ).getHeaders("foo");
+    }
+
+    @Test
+    public void headers() {
+        MatcherAssert.assertThat(
+            "Can't get the headers",
+            Collections.list(
+                new HttpServletRequestFake(
+                    new RqWithHeaders(
+                        new RqFake(),
+                        "TestHeader: someValue",
+                        "SomeHeader: testValue"
+                    )
+                ).getHeaders("testheader")
+            ),
+            Matchers.hasItems("someValue")
+        );
+    }
+
+    @Test
+    public void headerNames() {
+        MatcherAssert.assertThat(
+            "Can't get the header names",
+            Collections.list(
+                new HttpServletRequestFake(
+                    new RqWithHeaders(
+                        new RqFake(),
+                        "AnyHeader: anyValue",
+                        "CrazyHeader: crazyValue"
+                    )
+                ).getHeaderNames()
+            ),
+            Matchers.hasItems("host", "anyheader", "crazyheader")
+        );
+    }
+
+    @Test
+    public void method() {
+        MatcherAssert.assertThat(
+            "Can't get the request method",
+            new HttpServletRequestFake(new RqFake()).getMethod(),
+            new IsEqual<>("GET")
```

Comment:
@fabriciofx Please use `RqMethod.GET` instead of `"GET"` here

Reply:
@paulodamaso Done.